pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -350,6 +367,778 @@ var _ = Describe("VirtualMachine Mutator", func() {
 		}
 	})
 
+	Context("with InferFromVolume enabled", func() {
+
+		var (
+			pvc               *k8sv1.PersistentVolumeClaim
+			dvWithSourcePVC   *v1beta1.DataVolume
+			dvWithAnnotations *v1beta1.DataVolume
+			dsWithSourcePVC   *v1beta1.DataSource
+			dsWithAnnotations *v1beta1.DataSource
+		)
+
+		const (
+			inferVolumeName     = "inferVolumeName"
+			defaultInferedName  = "defaultInferedName"
+			defaultInferedKind  = "defaultInferedKind"
+			pvcName             = "pvcName"
+			dvWithSourcePVCName = "dvWithSourcePVC"
+			dsWithSourcePVCName = "dsWithSourcePVC"
+			dsWithAnnotionsName = "dsWithAnnotionsName"
+			unknownPVCName      = "unknownPVC"
+			unknownDVName       = "unknownDV"
+		)
+
+		BeforeEach(func() {
+			pvc = &k8sv1.PersistentVolumeClaim{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name:      pvcName,
+					Namespace: vm.Namespace,
+					Annotations: map[string]string{
+						apiinstancetype.DefaultInstancetypeAnnotation:     defaultInferedName,
```

Comment:
It might make sense to use different annotation values for different objects, so it is ensured the instancetype/preference is inferred from the intended combination of DataVolumeSource, DataVolumeTemplate, DataSource and PVC.